docs/configuration.md

```diff
@@ -0,0 +1,214 @@
+# Fissile Configuration
+Fissile requires some configuration to work with BOSH releases.  It is necessary
+to describe what docker images should be created; in general, each docker image
+(termed "role") will have one or more BOSH jobs.
+
+There are three main files for configuration: the [role manifest], [opinions],
+and [dark opinions].  A set of [environment files] will also be required.
+
+For examples of established configuration, please see [uaa-fissile-release.git]
+or [scf.git].  They are, respectively, configuration for UAA and a full
+(single-node) Cloud Foundry deployment.
+
+[role manifest]: #role-manifest
+[opinions]: #opinions-dark-opinions-and-environment
+[dark opinions]: #opinions-dark-opinions-and-environment
+[environment files]: #opinions-dark-opinions-and-environment
+[uaa-fissile-release.git]: https://github.com/SUSE/uaa-fissile-release
+[scf.git]: https://github.com/SUSE/scf/
+
+The rest of this document will also, as an example, describe how to create
+fissile configuration to place `nats` in a docker image.
+
+## Role Manifest
+The role manifest is the main configuration file for fissile.  It must contain a
+list of roles (analogous to BOSH VMs); each role will result in one docker
+image.  It can also have a configurable variables section to describe the
+tunable inputs, and a configuration templates section to map those variables to
+BOSH properties and related BOSHisms.  We will be working with the example role
+manifest for NATS:
+
+```yaml
+roles:
+- name: nats                       # The name of the role
+  jobs:                            # BOSH jobs this role will have
+  - name: nats
+    release_name: nats             # The name of the BOSH release this is from
+  tags:
```

Comment:
Nothing that's likely to be kept up to date; they're free-form.
Currently, we only have `clustered` and `dev-only` (which doesn't get exported to kube). I'd like us to drop `dev-only` instead, since we only do kube now.